### ISSUE TITLE
Different routes files with same package names collide

### ISSUE THREAD

**Issue by diamondo25** [playframework/playframework#6719]
### Play Version (2.5.x / etc)

Play 2.5.9

### Expected Behaviour

1. Create `conf/a.routes`, with contents `GET /index controllers.SomeController.aIndex`
2. Create `conf/b.routes`, with contents `GET /index controllers.SomeController.bIndex`
3. Add the following to `conf/routes`:
```
-> /a a.Routes
-> /b b.Routes
```
4. Reverse routing should be available for both `SomeController.aIndex` (/a/index) and `SomeController.bIndex` (/b/index)

### Actual Behavior

1. Create `conf/a.routes`, with contents `GET /index controllers.SomeController.aIndex`
2. Create `conf/b.routes`, with contents `GET /index controllers.SomeController.bIndex`
3. Add the following to `conf/routes`:
```
-> /a a.Routes
-> /b b.Routes
```
4. Play Routes compiler will generate `ReverseRoutes.scala` and `routes.java` under `target/scala-2.11/routes/main/controllers/` twice; once for `a.routes` and another time for `b.routes`, overriding the previous
5. Either `SomeController.aIndex` or `SomeController.bIndex` is not available

P.S. I am sorry for not strictly following the guide in the Issue screen; it did not fit the issue I was seeing.

**Comment by wsargent** [comment]
Can you provide a Play github project reproducing this issue?

Also, there is documentation on splitting the routes file available:

https://www.playframework.com/documentation/2.5.x/SBTSubProjects#Splitting-the-route-file

**Comment by diamondo25** [comment]
Thanks for your response. I'll set one up in not too long.

**Comment by diamondo25** [comment]
@wsargent Here you go: https://github.com/diamondo25/playframework-routes-bug

**Comment by mkurz** [comment]
Also see #8423, I think it's the same bug (but not 100% sure)

**Comment by diamondo25** [comment]
That is correct, the play router generator doesn't split per route file per classname.

**Comment by xuwei-k** [comment]
https://github.com/playframework/playframework/pull/10513

**Comment by ttymck** [comment]
After looking into this, I wanted to ask for feedback on my assessment of the situation; hopefully someone more familiar with the RoutesCompiler will be able to confirm if my understanding is correct:

1. The `compileRoutesFiles` SBT task passes along each individual routes file (`routes`, `a.routes`, `b.routes`) as separate tasks for compilation 
2. `RoutesCompiler.compile` operates on each individual compilation task from (1). 
    - The reverse router is output to a static file name (i.e. `ReverseRoutes.scala`) but **with a package prefix** (`controllers/`) of the routes contained within.
    - The **reverse router template** _groups the routes by controller name_ (`class ReverseSomeController`) within the same source file

As far as I can tell, there are two issues at play:
1. Since _two different tasks_ (one for `a.routes`, and another for `b.routes`) reference methods of controllers in the same package (the `controllers` package, in this case they both reference a method of `controllers.SomeController`), they both write to `target/scala2.11/routes/main/controllers`, overwriting it once
2. Even if we fixed the file overwrite/collision issue above, there would still be two conflicting `class ReverseSomeController` in `controllers.routes`. 

One possible solution would be to change `RoutesCompilerTask#file: File` to `files: Seq[File]` and change `RoutesCompiler.compile` to do something like:

```
val rules: List[Rule] = routesCompilerTask.files.flatMap(RoutesFileParser.parse(_).getOrElse(Nil))
```

Thus running one compilation task for _all_ routes files in the project.
